supabase/migrations/20230817110845_access_token.sql

```diff
@@ -0,0 +1,174 @@
+create table app.access_tokens(
+    id uuid primary key default gen_random_uuid(),
+    user_id uuid not null references auth.users(id),
+    token_hash bytea not null,
+    token_name text not null check (length(token_name) <= 64),
+    created_at timestamptz not null default now(),
+
+    unique (user_id, token_name)
+);
+
+grant insert
+    (id, user_id, token_hash, token_name)
+    on app.access_tokens
+    to authenticated;
+
+grant delete
+    on app.access_tokens
+    to authenticated;
+
+alter table app.access_tokens enable row level security;
+
+create policy access_tokens_select_policy
+    on app.access_tokens
+    as permissive
+    for select
+    to public
+    using ( auth.uid() = user_id );
+
+create policy access_tokens_insert_policy
+    on app.access_tokens
+    as permissive
+    for insert
+    to authenticated
+    with check ( auth.uid() = user_id );
+
+create policy access_tokens_delete_policy
+    on app.access_tokens
+    as permissive
+    for delete
+    to authenticated
+    using ( auth.uid() = user_id );
```

Comment:
May be easier for maintenance to have one permissive policy `for all` that enforces the `auth.uid() = user_id` binding, and then add restrictive policies if something needs stopping (like deletes or the such).

Reply:
If you look carefully, the topmost policy is for the `public` role and the other two are for `authenticated`. So, adding a `for all` policy and then restricting individual operations will need more policies: two `for all` policies, three policies for restricting `insert`, `update` and, `delete` to `public` and two policies for restricting `select` and `update` to `authenticated`. Which makes seven policies in total.